addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,874 @@
+     *
+     * The validation is made server-side to have the same test when we validate here and when we
+     * register the answer, but also to easily post a message as the bot ("Sorry, try again...").
+     *
+     * @private
+     */
+    _chatbotTriggerNextStep: async function () {
+        if (this._chatbotCurrentStep) {
+            if (this._chatbotCurrentStep.chatbot_step_type === 'question_email') {
+                let emailValidResult = await session.rpc('/im_livechat/chatbot_validate_email', {
+                    channel_uuid: this._livechat.getUUID(),
+                });
+
+                if (!emailValidResult.success) {
+                    // email is not valid, let the user try again
+                    this._chatbotEnableInput();
+                    if (emailValidResult.posted_message) {
+                        this._chatbotAddMessage(emailValidResult.posted_message);
+                    }
+                    return;
+                } else {
+                    this._chatbotCurrentStep.is_email_valid = true;
+                    this._chatbotSaveSession();
+                }
+            }
+        }
+
+        let result = await session.rpc('/im_livechat/chatbot_trigger_step', {
+            channel_uuid: this._livechat.getUUID(),
+            chatbot_script_id: this._chatbot.chatbot_script_id
+        });
+
+        if (result) {
+            if (result.chatbot_posted_message) {
+                this._chatbotAddMessage(result.chatbot_posted_message);
+            }
+
+            this._chatbotCurrentStep = result.chatbot_step;
+
+            this._chatbotProcessStep();
+        } else {
+            // did not find next step -> end the script
+            this._chatbotCurrentStep['chatbot_step_is_last'] = true;
+            this._renderMessages();
+            this._chatbotEndScript();
+        }
+
+        this._chatbotSaveSession();
+
+        return result;
+    },
+
+     //--------------------------------------------------------------------------
+     // Private - LiveChat Overrides
+     //--------------------------------------------------------------------------
+
+    /**
+     *
+     * @override
+     */
+     _askFeedback: function () {
+        this._super(...arguments);
+
+        this._chatWindow.$('.o_livechat_chatbot_end').addClass('d-none');
+        this._chatWindow.$('.o_composer_text_field')
+            .removeClass('d-none')
+            .val('');
+     },
+
+    /**
+     * Small override:
+     * If we resume an existing chatbot script, check after opening the chat window if we are on
+     * the last step, and "end script" if so.
+     *
+     * @private
+     * @override
+     */
+    _openChatWindow: function () {
+        return this._super(...arguments).then(() => {
+            if (!this._chatbotCurrentStep || !this._messages || this._messages.length === 0) {
+                return;
+            } else {
+                this._chatbotProcessStep();
+            }
```

Comment:
you can invert the condition and remove the else

```suggestion
            if (this._chatbotCurrentStep && this._messages && this._messages.length !== 0) {
                this._chatbotProcessStep();
            }
```